processing/src/main/java/org/apache/druid/query/JoinDataSource.java

```diff
@@ -96,6 +96,8 @@ public class JoinDataSource implements DataSource
   private final DimFilter leftFilter;
   @Nullable
   private final JoinableFactoryWrapper joinableFactoryWrapper;
+  @Nullable
+  private final JoinAlgorithm preferredJoinAlgorithm;
```

Comment:
rename to `joinAlgorithm`
note: I feel like that in reality this field will never be `null`

Reply:
The reason it is preferred is that if a hint is not supplied, it would be null, at least currently. Given that, should we still rename this to `joinAlgorithm`

Reply:
My mistake, I think it does pick up broadcast as the default at this level itself, so this change makes sense.

Reply:
Changed